drivers/rtc/rtc_nxp_irtc.c

```diff
@@ -0,0 +1,454 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_irtc
+
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/rtc.h>
+#include <zephyr/irq.h>
+
+struct nxp_irtc_config {
+	RTC_Type *base;
+	void (*irq_config_func)(const struct device *dev);
+	bool is_output_clock_enabled;
+	uint8_t clock_src;
+	uint8_t alarm_match_flag;
+};
+
+struct nxp_irtc_data {
+	bool is_dst_enabled;
+#ifdef CONFIG_RTC_ALARM
+	rtc_alarm_callback alarm_callback;
+	void *alarm_user_data;
+	uint16_t alarm_mask;
+	bool is_alarm_pending;
+#endif /* CONFIG_RTC_ALARM */
+};
+
+/* The IRTC Offset is 2112 instead of 1990 [2112 - 1990] -> [212] */
```

Comment:
the offset of the tm struct is `1900`, not `1990`. probably just a typo :)

Reply:
Thanks for noticing, I will fix this on the next update!